docs/source/configuration/authorization.mdx

```diff
@@ -644,7 +644,19 @@ This option will entirely reject queries that would have some parts filtered by
 ```yaml title="router.yaml"
 authorization:
   preview_directives:
+    enabled: true
     log_errors: true
 ```
 
-By default, when part of a query is filtered by authorization, the list of filtered paths is added to the response and logged by the router. Filtering parts of a query according to the client's rights may be seen as normal operation, and not warrant investigation by platform operators, so those logs can be disabled.
\ No newline at end of file
+By default, when part of a query is filtered by authorization, the list of filtered paths is added to the response and logged by the router. Filtering parts of a query according to the client's rights may be seen as normal operation, and not warrant investigation by platform operators, so those logs can be disabled.
+
+### dry_run
+
+```yaml title="router.yaml"
+authorization:
+  preview_directives:
+    enabled: true
+  dry_run: true # default: false
+```
+
+This option executes the authorization directives without modifying the query: it only generates the list of unauthorized paths and returns it as part of the response. This is a way to evaluate the impact of authorization policies without breaking existing traffic.
```

Comment:
Have you validated that returning top-level errors is what customers will expect with a `dry_run` option? Apollo Client defaults to raising exceptions when it sees top-level errors, so most client apps will fail.

When I see "dry_run" I assume that there will be no runtime effects on my applications. Maybe it should attach the filtered paths as extensions instead?

Reply:
the opinion from the client devs was that authorization errors could be easily ignored so it wouldn't be an issue. Even in normal operation, you can have a valid query for which there are authorization errors, that does not warrant raising the error up to the user

Reply:
will follow up in https://github.com/apollographql/router/issues/4116